### PR TITLE
Break the leak tracker code to packages.

### PR DESCRIPTION
**Pull request by polina-c**
Fixes https://github.com/dart-lang/sdk/issues/52367

### CODE REVIEW

File: pkgs/leak_tracker_flutter_test/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+dev_dependencies:
+  flutter:
+    sdk: flutter
+  flutter_lints: ^2.0.0
+  flutter_test:
+    sdk: flutter
+  integration_test:
+    sdk: flutter
+  leak_tracker: ^7.0.4
```
Comment (devoncarew) [inline pkgs/leak_tracker_flutter_test/pubspec.yaml:17]: Do you want to use a `path: ../leak_tracker` reference here? Or `leak_tracker: any` and then have a `dependency_overrides:` entry with `path: ../leak_tracker`; either will let you run against the repo version of leak_tracker.

File: pkgs/leak_tracker_testing/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: leak_tracker_testing
+version: 1.0.0
+description: Leak tracking code intended for usage in tests.
+repository: https://github.com/dart-lang/leak_tracker
+
+environment:
+  sdk: '>=3.0.0 <4.0.0'
+
+dependencies:
+  leak_tracker: ^7.0.4
```
Comment (devoncarew) [inline pkgs/leak_tracker_testing/pubspec.yaml:10]: Same comment here re: referencing the repo version of `leak_tracker`.

**Review by devoncarew — COMMENTED** [review]
Looks great - thanks for the refactor! Several comments in-line.

File: pkgs/leak_tracker_testing/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# 1.0.0
```
[the remark this answers is not in the document]
Comment (polina-c) [reply pkgs/leak_tracker_testing/CHANGELOG.md:1]: we are publishing it

File: pkgs/leak_tracker_testing/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: leak_tracker_testing
+version: 1.0.0
+description: Leak tracking code intended for usage in tests.
+repository: https://github.com/dart-lang/leak_tracker
```
[the remark this answers is not in the document]
Comment (polina-c) [reply pkgs/leak_tracker_testing/pubspec.yaml]: we are publishing it

File: pkgs/leak_tracker_flutter_test/pubspec.yaml (hunk 1 above)
Comment (polina-c) [reply pkgs/leak_tracker_flutter_test/pubspec.yaml:17]:
```
Publishable packages can't have 'path' dependencies.
Try adding a 'publish_to: none' entry to mark the package as not for publishing or remove the path dependency.
```

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -25,25 +25,44 @@ jobs:
       - name: version
         run: flutter --version
 
-      - name: flutter pub get
+      - name: pub get root
```
[the remark this answers is not in the document]
Comment (polina-c) [reply .github/workflows/ci.yaml:28]: It seems different jobs run on different boxes and it does not make sense to separately install flutter for each package as it is expensive.

File: pkgs/leak_tracker_testing/CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (devoncarew) [reply pkgs/leak_tracker_testing/CHANGELOG.md:1]: Do we expect people to use it themselves? If no, we should update the readme to clarify its use and intent, and update the package on pub.dev to be unlisted.

File: pkgs/leak_tracker_flutter_test/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+name: leak_tracker_flutter_test
+version: 0.0.1
+description: Tests for leak_tracker that depend on Flutter framework.
+repository: https://github.com/dart-lang/leak_tracker
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply pkgs/leak_tracker_flutter_test/pubspec.yaml]: Same as for `leak_tracker_testing` - we should update the readme and make sure the package is unlisted.

File: .github/workflows/ci.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (devoncarew) [reply .github/workflows/ci.yaml:28]: FWIW, some sharding of testing is not uncommon for other packages.

[the remark this answers is not in the document]
Comment (polina-c) [reply .github/workflows/ci.yaml:28]: Yes, there are options.
For this case I chose not to contribute to global worming more :)

File: pkgs/leak_tracker_testing/pubspec.yaml (hunk 2 above)
Comment (devoncarew) [reply pkgs/leak_tracker_testing/pubspec.yaml:10]: Ah, ok. I'm not familiar w/ why this is published, but given that it is, try:

for the pubspec:

```yaml
dependencies:
  leak_tracker: ^7.0.4
```

and create a file next to it called `pubspec_overrides.yaml`:

```yaml
dependency_overrides:
  leak_tracker:
    path: ../leak_tracker
```

Comment (polina-c) [reply pkgs/leak_tracker_testing/pubspec.yaml:10]: it worked inside the same file

File: pubspec.yaml [hunk 7]
```diff
@@ -1,28 +1,9 @@
-name: leak_tracker
-version: 7.0.4
-description: A framework for memory leak tracking for Dart and Flutter applications.
+name: leak_tracker_root
```
[the remark this answers is not in the document]
Comment (polina-c) [reply pubspec.yaml]: Yes, bots pass, but my VSCode gives me warning:

Screenshot 2023-06-06 at 5 05 34 PM

[the remark this answers is not in the document]
Comment (polina-c) [reply pubspec.yaml:1]: filed a bug: https://github.com/Dart-Code/Dart-Code/issues/4574

File: pkgs/leak_tracker_testing/CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (polina-c) [reply pkgs/leak_tracker_testing/CHANGELOG.md:1]: Yes, people will use it to test their packages.

**Review by devoncarew — APPROVED** [review]
We should also retain a README.md file for the top level of the repo.

**Comment by polina-c** [comment]
> We should also retain a README.md file for the top level of the repo.

It is here: https://github.com/dart-lang/leak_tracker/blob/main/README.md

**Pull request merged**